### PR TITLE
[2.1.x] Sort label by key #1142

### LINKED ISSUE #1142: Operator labels listed in metadata.annotation change order randomly
The CR is updated unnecessarily due to the subject.
This is due to the fact that iteration over a map has undefined order in golang.

### PR DESCRIPTION
**Pull request by rigazilla**
fixes #1142

### CODE REVIEW

File: pkg/apis/infinispan/v1/types_util.go [hunk 1]
```diff
@@ -592,13 +592,20 @@ func applyLabels(ispn *Infinispan, envvar, annotationName string) error {
...
 	if err == nil {
 		if len(labelMap) > 0 {
-			svcLabels := ""
 			if ispn.Labels == nil {
 				ispn.Labels = make(map[string]string)
 			}
-			for name, value := range labelMap {
-				ispn.Labels[name] = value
-				svcLabels = svcLabels + name + ","
+			keys := make([]string, len(labelMap))
```
Comment (dmvolod) [inline pkg/apis/infinispan/v1/types_util.go:598]: @rigazilla can you create unit test for this functionality?

Comment (rigazilla) [reply pkg/apis/infinispan/v1/types_util.go:598]: @dmvolod let me see

Comment (rigazilla) [reply pkg/apis/infinispan/v1/types_util.go:598]: @dmvolod I've added some tests

**Review by dmvolod — APPROVED** [review]

File: pkg/apis/infinispan/v1/types_util.go (hunk 1 above)
Comment (dmvolod) [reply pkg/apis/infinispan/v1/types_util.go:598]: Thanks, @rigazilla 
Looks good, but CI seems to be hangs

Comment (dmvolod) [reply pkg/apis/infinispan/v1/types_util.go:598]: Linter was fails. Please verify

**Pull request merged**